File: app/src/main/java/com/butterflymx/butterflymxapiclient/utils/mvp/BasePresenter.java
package com.butterflymx.butterflymxapiclient.utils.mvp;

public abstract class BasePresenter<T extends BaseView> implements MvpPresenter<T> {

    private T view;

    @Override
    public void attachView(T mvpView) {
        view = mvpView;
    }

    @Override
    public void detachView() {
        view = null;
    }

    public T getView() {
        return view;
    }

    protected boolean isViewAttached() {
        return view != null;
    }

    @Override
    public void destroy() {
    }

    @Override
    public void viewIsReady() {
    }


}

File: app/src/main/java/com/butterflymx/butterflymxapiclient/utils/mvp/BasePresenter.kt
package com.butterflymx.butterflymxapiclient.utils.mvp

abstract class BasePresenter<T : BaseView> : MvpPresenter<T> {

    var view: T? = null
        private set

    protected val isViewAttached: Boolean
        get() = view != null

    override fun attachView(mvpView: T) {
        view = mvpView
    }

    override fun detachView() {
        view = null
    }

    override fun destroy() {}

    override fun viewIsReady() {}


}